awx/main/scheduler/dependency_graph.py

```diff
@@ -72,7 +72,12 @@ def project_update_blocked_by(self, job):
         return self.get_item(self.PROJECT_UPDATES, job.project_id)
 
     def inventory_update_blocked_by(self, job):
-        return self.get_item(self.INVENTORY_SOURCE_UPDATES, job.inventory_source_id)
+        inventory_source_block = self.get_item(self.INVENTORY_SOURCE_UPDATES, job.inventory_source_id)
+        project_block = self.get_item(self.PROJECT_UPDATES, job.inventory_source.source_project_id)
+        return inventory_source_block or project_block
+
+    def inventory_source_update_blocked_by(self, job):
+        return self.get_item(self.PROJECT_UPDATES, job.source.project_id)
```

Comment:
I'm not reading this coherently. Is `job` here an inventory update, or a job?

Also, the `InventoryUpdate` field that you want would be `source_project` IIRC, not `source`, so I'd expect to see reference to `.source_project_id`, but not totally sure what the thought here is.

Reply:
I mistakenly left this method in, the only method we need to update is `inventory_update_blocked_by`